### PR TITLE
Perform a describe stacks call only when a post install script has been provided

### PR DESCRIPTION
**Pull request by ermanno**
### Description of changes
Perform a describe stacks call only when a post install script has been provided.

### Test
Manually tested by launching an update after manually modifying the `fetch_and_run` script on the head node.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-config/templates/default/init/fetch_and_run.erb [hunk 1]
```diff
@@ -48,40 +48,34 @@ function get_stack_status () {
...
-  if [ "${stack_status}" != "UPDATE_IN_PROGRESS" ]; then
-    echo "Post update hook called with CFN stack in state ${stack_status}, doing nothing"
-    exit 0
-  fi
-
   if [ "${cfn_postupdate}" != "NONE" ]; then
-    file="${cfn_postupdate}"
-    if [ "${cfn_postupdate_args}" != "NONE" ]; then
-      download_run ${cfn_postupdate} "${cfn_postupdate_args[@]}"
+    stack_status=$(get_stack_status) || error_exit "Failed to get the stack status, check the HeadNode instance profile's Iam policies"
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-config/templates/default/init/fetch_and_run.erb]: nitpick, IAM should probably be all capitals, [example in the docs](https://docs.aws.amazon.com/parallelcluster/latest/ug/HeadNode-v3.html#yaml-HeadNode-Iam-S3Access:~:text=Names%20(ARNs)%20of-,IAM,-policies%20for%20Amazon)

**Review by eantonin — DISMISSED** [review]

**Review by eantonin — APPROVED** [review]
LGTM

**Review by enrico-usai — COMMENTED** [review]
You removed `file` env variable, but it is used in the line 39.

**Comment by ermanno** [comment]
We can use `url` instead of `file` since it has the same content. PR: https://github.com/aws/aws-parallelcluster-cookbook/pull/1611

**Pull request merged**